docs/reference/python/virtual-environments.md

```diff
@@ -0,0 +1,246 @@
+between the two:
+
+| Virtualenv                                        | Anaconda                                                 |
+| :------------------------------------------------ | :------------------------------------------------------- |
+| Environment files are local.                      | Environment files are available globally.                |
+| Must activate environment by giving the path.     | Can activate the environment without knowing the path.   |
+| Can only use `pip` to install packages.           | Can either use `pip` or built-in `conda` package manager.|
+| Installation is very simple.                      | Installation takes more effort.                          |
+| Can only install python packages.                 | In addition to packages, you can download many data science tools.|
+
+We recommend **virtualenv** over Anaconda because of its simplicity. However, feel free to appeal to your preferences.
+
+## Installation
+
+=== "Virtualenv"
+
+    If you already have python and the pip package manager installed, just execute the following:
+
+    ```bash title="Using pip to install virtualenv"
+    pip install virtualenv
+    ```
+
+=== "Anaconda"
+
+    Go to the official [Anaconda website](https://www.anaconda.com/) and follow the installation instructions 
+    for your operating system.
+
+## Using virtual environments
+
+The name of a virtual environment is configurable. For the purposes of this site, we will use `env` as the environment
+name unless specified otherwise.
+
+### Creating a virtual environment
+
+=== "Virtualenv"
+
+    Since virtualenv creates the environment directory in a specific location, make sure that you
+    are in the located in the project that you want to work on.
+
+    ```bash title="Create virtual environment with virtualenv"
+    # Go to desired location
+    cd <PATH TO DIRECTORY>
+
+    # Create the environment with the name env
+    python3 -m venv env
+    ```
+
+    Verify that your environment is created by examining your current directory and look for the directory
+    that matches the name of your virtual environment.
+
+=== "Anaconda"
+
+    Since the environment will be available globally, there is no need to go to a specific location to create
+    it.
+
+    ``` bash title="Create virtual environment with Anaconda"
+    # Create environment with name env and python version
+    conda env create -n env python=<PYTHON VERSION NUM>
+    ```
+
+    If you don't specify a python version, the default is the latest version. Verify that your environment is created
+    by executing `conda env list`.
+
+### Activating the virtual environment
+
+To use the virtual environment, you must activate it.
+
+=== "Virtualenv"
+
+    ```batch title="Activation for Windows"
+    env\Scripts\activate
+    ```
+
+    ```bash title="Activation for MacOS or Linux"
+    source env/bin/activate
+    ```
```

Comment:
You could make tabs for the different OS's

Reply:
Ok. I also think I wanna add some of those fancy icons in the tabs :)

I'll look into that

Reply:
You can search for icons [here](https://squidfunk.github.io/mkdocs-material/reference/icons-emojis/#search)

Reply:
Resolved by addressing dependency issues.